app/services/messaging/bsnl/api.rb

```diff
@@ -0,0 +1,35 @@
+class Messaging::Bsnl::Api
+  HOST = "bulksms.bsnl.in"
+  PORT = 5010
+
+  def self.get_template_details
+    post("/api/Get_Content_Template_Details")["Content_Template_Ids"]
+  end
+
+  def self.credentials
+    {
+      header: ENV["BSNL_IHCI_HEADER"],
+      message_type: "SI",
+      entity_id: ENV["BSNL_IHCI_ENTITY_ID"],
+      jwt: ENV["BSNL_JWT_TOKEN"]
+    }
+  end
+
+  def self.post(path, body = nil)
+    raise Messaging::Error "Missing BSNL credentials" unless credentials.values.all?
```

Comment:
This exception probably belongs in the `initializer`, putting here so we don't miss it later.